File: tools/LogViewer/test/hello.cpp
#include <windows.h>
#include <string>
#include <iostream>

int main()
{
    // 打开管道（宽字符版本）
    HANDLE pipe = CreateFileW(
        L"\\\\.\\pipe\\LogWindowPipe", // 宽字符管道名
        GENERIC_WRITE,
        0, nullptr, OPEN_EXISTING, 0, nullptr
    );

    if (pipe == INVALID_HANDLE_VALUE)
    {
        std::wcerr << L"Failed to open pipe\n";
        return 1;
    }

    // 构造要发送的日志（wchar_t）
    std::wstring log = L"这是中文日志 🌟\n";

    DWORD written;
    BOOL ok = WriteFile(
        pipe,
        log.c_str(),
        static_cast<DWORD>(log.size() * sizeof(wchar_t)), // 注意单位是字节
        &written,
        nullptr
    );

    if (!ok)
    {
        std::wcerr << L"WriteFile failed\n";
        CloseHandle(pipe);
        return 1;
    }

    CloseHandle(pipe);
    std::wcout << L"Log sent successfully\n";
    return 0;
}

File: tools/LogViewer/test/hello.zig
const std = @import("std");
const c = @cImport({
    @cInclude("windows.h");
});

pub fn main() !void {
    const allocator = std.heap.page_allocator;

    const utf8_log = "这是中文日志 🌟\n";

    var log_utf16: [:0]u16 = try std.unicode.utf8ToUtf16LeAllocZ(allocator, utf8_log);
    defer allocator.free(log_utf16);

    const pipe_name_utf8 = "\\\\.\\pipe\\LogWindowPipe";
    var pipe_name: [:0]u16 = try std.unicode.utf8ToUtf16LeAllocZ(allocator, pipe_name_utf8);
    defer allocator.free(pipe_name);

    const pipe: c.HANDLE = c.CreateFileW(
        pipe_name.ptr,
        c.GENERIC_WRITE,
        0,
        null,
        c.OPEN_EXISTING,
        0,
        null,
    );

    if (pipe == c.INVALID_HANDLE_VALUE) {
        std.debug.print("Failed to open pipe\n", .{});
        return;
    }

    var bytes_written: c.DWORD = 0;
    const ok = c.WriteFile(
        pipe,
        @ptrCast(log_utf16.ptr),
        @intCast(log_utf16.len * @sizeOf(u16)), // 字节数
        &bytes_written,
        null,
    );

    if (ok == 0) {
        std.debug.print("WriteFile failed\n", .{});
        _ = c.CloseHandle(pipe);
        return;
    }

    _ = c.CloseHandle(pipe);
    std.debug.print("Log sent successfully\n", .{});
}